applications/FluidDynamicsHydraulicsApplication/python_scripts/navier_stokes_two_fluid_hydraulic_fractional_solver.py

```diff
@@ -0,0 +1,644 @@
+
+
+
+        # STEP II: Convect the free surface according to the fractional velocity 
+
+        # Before doing this second step, the fractional velocity data is copied to the velocity data since the level set convection process takes velocity variable as convection variable. 
+        # And the previous previous velocity is copied in an auxiliar variable
+        KratosMultiphysics.VariableUtils().CopyModelPartNodalVar(KratosCFD.FRACTIONAL_VELOCITY,KratosMultiphysics.VELOCITY, self.main_model_part, self.main_model_part, 0, 0)
+        KratosMultiphysics.VariableUtils().CopyModelPartNodalVar(KratosMultiphysics.VELOCITY,KratosCFD.AUXILIAR_VECTOR_VELOCITY, self.main_model_part, self.main_model_part, 0, 0)
+
+        self.__PerformLevelSetConvection()
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Level-set convection is performed.")
+
+        # After the convection process, the velocity is copied back to the original state. 
+        KratosMultiphysics.VariableUtils().CopyModelPartNodalVar(KratosCFD.AUXILIAR_VECTOR_VELOCITY,KratosMultiphysics.VELOCITY, self.main_model_part, self.main_model_part, 0, 0)
+
+        # Perform distance correction to prevent ill-conditioned cuts
+        self._GetDistanceModificationProcess().ExecuteInitializeSolutionStep()
+
+        # Update the DENSITY and DYNAMIC_VISCOSITY values according to the new level-set
+        self._SetNodalProperties()
+
+        # Accumulative water volume error ratio due to level set. Adding source term
+        self._ComputeVolumeError()
+
+        if self.artificial_viscosity:
+            self.__CalculateArtificialViscosity()
+
+        # Initialize the solver current step
+        self._GetSolutionStrategy().InitializeSolutionStep()
+
+        # We set this value at every time step as other processes/solvers also use them
+        # Note that this is required as the convection processes may set a different value (this is the one to be used in the Navier-Stokes element)
+        dynamic_tau = self.settings["formulation"]["dynamic_tau"].GetDouble()
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, dynamic_tau)
+
+
+    def FinalizeSolutionStep(self):
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Mass and momentum conservation equations are solved.")
+
+        # Recompute the distance field according to the new level-set position
+        if self._reinitialization_type != "none":
+            self._GetDistanceReinitializationProcess().Execute()
+            KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Redistancing process is finished.")
+
+        # Prepare distance correction for next step
+        self._GetDistanceModificationProcess().ExecuteFinalizeSolutionStep()
+        
+        self._GetSolutionStrategy().FinalizeSolutionStep()
+
+        if self.outlet_inflow_limiter:
```

Comment:
```suggestion
        # Correct current step solution to prevent inflow through outlets
        if self.outlet_inflow_limiter:
```

Reply:
In the end, it was modified and is now an option within the hydraulic boundary conditions. Therefore I am revoming from the solver.